src/shims/unix/fs.rs

```diff
@@ -967,26 +965,21 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
             builder.mode(mode);
         }
 
-        let result = builder.create(path).map(|_| 0i32);
-
-        Ok(Scalar::from_i32(this.try_unwrap_io_result(result)?))
+        Ok(builder.create(path))
     }
 
-    fn rmdir(&mut self, path_op: &OpTy<'tcx>) -> InterpResult<'tcx, Scalar> {
+    fn rmdir(&mut self, path_op: &OpTy<'tcx>) -> InterpResult<'tcx, io::Result<()>> {
         let this = self.eval_context_mut();
 
         let path = this.read_path_from_c_str(this.read_pointer(path_op)?)?;
 
         // Reject if isolation is enabled.
         if let IsolatedOp::Reject(reject_with) = this.machine.isolated_op {
             this.reject_in_isolation("`rmdir`", reject_with)?;
-            this.set_last_error_from_io_error(ErrorKind::PermissionDenied.into())?;
-            return Ok(Scalar::from_i32(-1));
+            return Ok(Err(ErrorKind::PermissionDenied.into()));
         }
 
-        let result = remove_dir(path).map(|_| 0i32);
-
-        Ok(Scalar::from_i32(this.try_unwrap_io_result(result)?))
+        Ok(remove_dir(path))
     }
 
     fn opendir(&mut self, name_op: &OpTy<'tcx>) -> InterpResult<'tcx, Scalar> {
```

Comment:
Why does this one still return `Scalar`?

Reply:
It (and others) return error codes via `this.eval_libc("EACCES");` without a corresponding `ErrorKind` existing in libcore. I plan to use an error enum that wraps io errors and has variants for all the eval_libc errors we use in miri. May turn out overkill, but I wanna see how it looks before discarding it

Reply:
Seems easier to use `Scalar` as the universal type for what to store in errno -- we can already convert both `io::Error` and libc constants into `Scalar`.